homeassistant/components/plugwise/__init__.py

```diff
@@ -1 +1,158 @@
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+# SENSOR_PLATFORMS = ["sensor"]
+# ALL_PLATFORMS = ["binary_sensor", "climate", "sensor", "switch"]
+ALL_PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Plugwise platform."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Plugwise Smiles from a config entry."""
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = Smile(
+        host=entry.data["host"], password=entry.data["password"], websession=websession
+    )
+
+    try:
+        connected = await api.connect()
+
+        if not connected:
+            _LOGGER.error("Unable to connect to Smile")
+            raise ConfigEntryNotReady
+
+    except Smile.PlugwiseError:
+        _LOGGER.error("Error while communicating to device")
+        raise ConfigEntryNotReady
+
+    except asyncio.TimeoutError:
+        _LOGGER.error("Timeout while connecting to Smile")
+        raise ConfigEntryNotReady
+
+    if api.smile_type == "power":
+        update_interval = timedelta(seconds=10)
+    else:
+        update_interval = timedelta(seconds=60)
+
+    async def async_update_data():
+        """Update data via API endpoint."""
+        try:
+            async with async_timeout.timeout(10):
+                await api.full_update_device()
+                return True
+        except Smile.XMLDataMissingError:
+            raise UpdateFailed("Smile update failed")
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="Smile",
+        update_method=async_update_data,
+        update_interval=update_interval,
+    )
+
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    api.get_all_devices()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "api": api,
+        "coordinator": coordinator,
+    }
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, api.gateway_id)},
+        manufacturer="Plugwise",
+        name=entry.title,
+        model=f"Smile {api.smile_name}",
+        sw_version=api.smile_version[0],
+    )
+
+    # single_master_thermostat = api.single_master_thermostat()
+    platforms = ALL_PLATFORMS
+    # if single_master_thermostat is None:
+    #     platforms = SENSOR_PLATFORMS
+
+    for component in platforms:
```

Comment:
```suggestion
    # single_master_thermostat = api.single_master_thermostat()
    platforms = ALL_PLATFORMS
    # if single_master_thermostat is None:
    #     platforms = SENSOR_PLATFORMS

    for component in ALL_PLATFORMS:
```

Reply:
I left it in half-purposely as to show you why the consts where defined, this is neccesary once we add the sensor (and forthcoming binary_sensor and switch) platforms. I'll remove both (your suggestion and the other comments) for keeping this PR pure.